test/CadlRanchProjects.Tests/type-property-nullable.cs

```diff
@@ -285,5 +289,175 @@ public Task Type_Property_Nullable_CollectionsString_patchNull() => Test(async (
             var response = await new NullableClient(host, null).GetCollectionsStringClient().PatchNullAsync(RequestContent.Create(value));
             Assert.AreEqual(204, response.Status);
         });
+
+        [Test]
+        public void RequiredNullableAreNotWriteable()
+        {
+            var requiredStringList = TypeAsserts.HasProperty(typeof(StringProperty), nameof(StringProperty.NullableProperty), BindingFlags.Public | BindingFlags.Instance);
+
+            Assert.Null(requiredStringList.SetMethod);
+        }
+
+        [Test]
+        public void RequiredNullableAreNotSetable()
+        {
+            var requiredStringList = TypeAsserts.HasProperty(typeof(StringProperty), nameof(StringProperty.NullableProperty), BindingFlags.Public | BindingFlags.Instance);
+
+            Assert.Null(requiredStringList.SetMethod);
+        }
+
+        [Test]
+        public void NotRequiredNullableListsSerializedAsNull()
```

Comment:
how about we say optional?
```suggestion
        public void OptionalNullableListsSerializedAsNull()
```

Reply:
The test name is not aligned with the scenario.
In the spec, the `nullableProperty` property in this model is required.
This should be
```suggestion
        public void RequiredNullableListsSerializedAsNull()
```

Reply:
fixed